### PR TITLE
Align custom traffic action with osc

### LINKED ISSUE #522: Inconsistency between OSC and OSI for custom traffic actions
#### Describe the bug
Both OSC 1.0 and OSI provide definitions for custom actions to be specified.  These allow for a user to configure custom strings in a scenario which are then passed via OSI at runtime to the system under test.

In OSC it is possible to specify two string values (type and content) and in OSI only one (command).

#### Describe how to reproduce the bug
OSC documentation: https://releases.asam.net/OpenSCENARIO/1.0.0/Model-Documentation/content/CustomCommandAction.html

OSI definition: https://github.com/OpenSimulationInterface/open-simulation-interface/blob/master/osi_trafficcommand.proto#L484

For clarity, in OSC you might see XML values like:

* `<CustomCommandAction type="manoeuvre">ExitHighway</CustomCommandAction>`
* `<CustomCommandAction type="sensor-failure">FrontRightCamera</CustomCommandAction>`

#### Describe the expected behavior
They should be consistent.

### PR DESCRIPTION
**Pull request by caspar-ai**
﻿#### Reference to a related issue in the repository
Closes #522

#### Add a description
Add a second string field to the custom action to mirror that in the OSC definitions.

In line with standard OSI the field is optional so this is not a breaking change.

#### Take this checklist as orientation for yourself, if this PR is ready for the Change Control Board:
- [x] My suggestion follows the [style and contributors guidelines](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/howtocontribute.html).
- [x] I have taken care about the [documentation](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/commenting.html).
- [x] I have done the [DCO signoff](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/howtocontribute.html#developer-certification-of-origin-dco).
- [x] My changes generate no errors when passing CI tests. 
- [x] I have successfully implemented and tested my fix/feature locally.
- [x] Appropriate reviewer(s) are assigned.

### CODE REVIEW

**Review by clemenshabedank — APPROVED** [review]

File: osi_trafficcommand.proto [hunk 1]
```diff
@@ -490,6 +490,10 @@ message TrafficAction
         // The custom command given to the traffic participant.
         //
         optional string command = 2;
+
+        // The type of the custom command given to the traffic participant.
+        //
+        optional string command_type = 3;
```
Comment (kmeids) [inline osi_trafficcommand.proto:496]: **Output CCB 09.06.2021:**
1. Please provide more documentation on the "command" and "command_type" messages, stating the difference between them and how/in which situations should it be used.
2. As a  "note", please mention for both commands that they are mapped to OSC.

**Review by kmeids — COMMENTED** [review]
Please check comments

File: osi_trafficcommand.proto (hunk 1 above)
Comment (caspar-ai) [reply osi_trafficcommand.proto]: Thanks @kmeids I have updated the comments as suggested.

Does that look OK to you?

**Comment by caspar-ai** [comment]
@kmeids any update from the CCB group about whether this can be merged?

**Comment by kmeids** [comment]
**CCB Output 21.07.2021**
1. @max-rosin please do a documentation check.
2. @pmai to merge after documentation check is done.

File: osi_trafficcommand.proto [hunk 2]
```diff
@@ -487,9 +487,21 @@ message TrafficAction
         //
         optional ActionHeader action_header = 1;
 
-        // The custom command given to the traffic participant.
+        // The custom command given to the traffic participant.  Used to convey a specific instruction
+        // (for example "exit_highway"), or event (for example "left_indicator_activated").
+        //
+        // \note This corresponds to the content of the OpenSCENARIO 1.0 CustomCommandAction field.
```
Comment (max-rosin) [inline osi_trafficcommand.proto:493]:
```suggestion
        // The custom command given to the traffic participant. Used to convey a specific instruction
        // (for example, "exit_highway"), or event (for example, "left_indicator_activated").
        //
        // \note This corresponds to the content of the OpenSCENARIO 1.0 CustomCommandAction field.
```

File: osi_trafficcommand.proto [hunk 3]
```diff
@@ -487,9 +487,21 @@ message TrafficAction
...
         //
         optional string command = 2;
+
+        // The type of the custom command given to the traffic participant.  Can be used to simplify
+        // how commands are grouped, for example the command_type could be "sensor_failure" and the
+        // command value (above) could be "front_right_camera".  This prevents needing long commands
+        // (like "sensor_failure: front_right_camera").
+        //
+        // \note This corresponds to the "type" attribute of the OpenSCENARIO 1.0 CustomCommandAction.
+        //
```
Comment (max-rosin) [inline osi_trafficcommand.proto:503]:
```suggestion
        // The type of the custom command given to the traffic participant. Can be used to simplify
        // how commands are grouped. For example, the command_type could be "sensor_failure" and the
        // command value could be "front_right_camera".  This avoids long commands, like
        //"sensor_failure: front_right_camera".
        //
        // \note This corresponds to the "type" attribute of the OpenSCENARIO 1.0 CustomCommandAction.
        //
```

**Comment by caspar-ai** [comment]
@max-rosin I agreed with your comment suggests and have applied those changes.  Is there anything else to do on the documentation side?

@pmai if not, I think over to you to merge.

**Pull request merged**